sdk/servicebus/service-bus/src/core/streamingReceiver.ts

```diff
@@ -402,9 +403,49 @@ export class StreamingReceiver extends MessageReceiver {
     await this._receiverHelper.suspend();
   }
 
-  async init(useNewName: boolean, abortSignal?: AbortSignalLike): Promise<void> {
-    const options = this._createReceiverOptions(useNewName, this._getHandlers());
-    await this._init(options, abortSignal);
+  async init(
+    // TODO: make onError? required
+    args: { useNewName: boolean; connectionId: string; onError?: OnError } & Pick<
+      OperationOptionsBase,
+      "abortSignal"
+    >
+  ) {
+    const config: RetryConfig<void> = {
+      operation: async () => {
+        try {
+          return this._initOnce(args);
+        } catch (err) {
+          args.onError &&
+            args.onError({
+              error: err,
+              errorSource: "receive",
+              entityPath: this.entityPath,
+              fullyQualifiedNamespace: this._context.config.host
+            });
+
+          // once we're at this point where we can spin up a connection we're past the point
+          // of fatal errors like the connection string just outright being malformed, for instance.
+          err.retryable = true;
+        }
+      },
+      connectionId: args.connectionId,
+      operationType: RetryOperationType.receiverLink,
+      retryOptions: {
+        ...this._retryOptions,
```

Comment:
Should we even be respecting the user provided retry options at this point? I see that we change the maxTries, but why bother respecting the retry mode or interval between retries or timeouts if this is meant to be something we control and want to try forever?

Reply:
I'm on the fence about this. I'm not well into the mindset of users on this.

I'll check in with .net on this - I'm curious how they handle it. I seem to recall they actually did incorporate the retry argument into their retries, even when doing 'forever'.

Reply:
I changed this to be different than both of us had guessed.

I decided to think of the retries as being an infinite series of "cycles" - the user's retry policy will cover each cycle (basically, a single call of retry<T>). This allows them to control how long they'd like to wait in between retries of init, etc... in case control over that is interesting to them.

However, after a set of retries is exhausted we will now head directly into the next retry<T> cycle, infinitely.

I feel like this is a decent compromise to give the user some control while still providing the infinite retry that we ultimately want.